backend/src/main/kotlin/org/loculus/backend/service/files/S3Service.kt

```diff
@@ -0,0 +1,115 @@
+package org.loculus.backend.service.files
+
+import io.minio.GetPresignedObjectUrlArgs
+import io.minio.MinioClient
+import io.minio.SetObjectTagsArgs
+import io.minio.http.Method
+import org.loculus.backend.config.S3BucketConfig
+import org.loculus.backend.config.S3Config
+import org.springframework.stereotype.Service
+import java.util.concurrent.TimeUnit
+
+private const val PRESIGNED_URL_EXPIRY_SECONDS = 60 * 30
```

Comment:
This should probably be configurable in values.yaml

Reply:
Possibly, although not at the moment in this PR; we will also touch the expiry time again in #4050, so I've added a sentence there for now